sw/device/silicon_creator/lib/drivers/uart.c

```diff
@@ -97,14 +97,21 @@ int uart_getchar(uint32_t timeout_ms) {
   return n ? (int)ch : -1;
 }
 
-size_t uart_write(const uint8_t *data, size_t len) {
-  size_t total = len;
+void uart_write(const void *data, size_t len) {
+  const uint8_t *d = (const uint8_t *)data;
   while (len) {
-    uart_putchar(*data);
-    data++;
+    uart_putchar(*d++);
     len--;
   }
-  return total;
+}
+
+void uart_write_hex(uint32_t val, size_t len) {
+  static const uint8_t kHexTable[16] = "0123456789abcdef";
+  size_t i = len * 8;
```

Comment:
Here it would also be good to constraint the length up to word size.

Reply:
Done, although this is low risk: no pointer is followed, so corrupting the length will just result in a bunch of extra `00`s being sent to the UART.